Reorganize READMEs and make each package's directions clearer (#24)
foxglove/ws-protocol: typescript/ws-protocol-examples/README.md

# Foxglove WebSocket examples in TypeScript

> The Foxglove WebSocket protocol is encoding-agnostic, as long as the desired encoding is supported by both client and server. It currently supports Protobuf and JSON messages.

Reviewer: I think I intentionally made this intro paragraph a bit different from the others, linking out to the main docs, since it's meant to be an example showcase rather than description of the protocol.

## Installation

Install the `ws-protocol-examples` `npm` package:

```
$ npm install @foxglove/ws-protocol-examples

> ```

Reviewer: There is no required installation step for this package.

Run example servers `sysmon` and `image-server` using the following commands:

```
$ npx @foxglove/ws-protocol-examples sysmon
$ npx @foxglove/ws-protocol-examples image-server
```

To see the data transmitted by one of these servers, open [studio.foxglove.dev](https://studio.foxglove.dev) in a browser, and initiate a Foxglove WebSocket connection to the respective WebSocket URL.

<img width="500" alt="Foxglove Studio displaying memory and CPU usage from the system monitor example" src="https://user-images.githubusercontent.com/14237/145313065-85c05645-6b29-4eb2-a498-849c83f8792d.png">

You can also run a simple example client that subscribes to messages on all channels with the `json` encoding.

```
$ npx @foxglove/ws-protocol-examples simple-client localhost:8765
```

## Development

> The `ws-protocol-examples` package lives inside a monorepo that uses [yarn workspaces](https://yarnpkg.com/features/workspaces), so most commands (other than `yarn install`) should be prefixed with `yarn workspace @foxglove/ws-protocol-examples...`.

Reviewer: Space was intentional, though we could remove the `...` if you find it strange. Just wanted to show that it is meant to be a prefix on other commands.

# Foxglove Studio WebSocket protocol examples
# Foxglove WebSocket examples

This package provides example server and client scripts using the [Foxglove Studio WebSocket protocol](https://github.com/foxglove/ws-protocol).
The [Foxglove WebSocket protocol](https://github.com/foxglove/ws-protocol) supports any data encoding, as long as it is supported by both client and server. This package provides example server and client implementation of the protocol.

## Usage
## Example servers

Run the `sysmon` example server, which uses JSON-encoded data to transmit system monitoring information like memory and CPU usage:

```

> $ npx @foxglove/ws-protocol-examples --help

Reviewer: Do you think we might want to keep the `--help` line, maybe as italic/quoted text at the top of this section, just to help people discover more options for themselves?

Author: Added to introduction instead.

> Run the `image-server` example server, which uses JSON-encoded data to transmit images:

Reviewer: Not really a big deal, but I think if you try to run both servers at once, the second one will fail because the port is already in use. Wonder if there is anything we can do with the wording/structure here to subtly encourage people to run one at a time.

Author: ```
You must close out each server (`control` + `c`) before starting up the other.

To see data from either server, open [Foxglove Studio](https://studio.foxglove.dev?ds=foxglove-websocket&ds.url=ws://localhost:8765/) with a Foxglove WebSocket connection to `ws://localhost:8765/`:
```

# Foxglove WebSocket examples

> This package provides example server and client implementation of the [Foxglove WebSocket protocol](https://github.com/foxglove/ws-protocol).

Reviewer: ```suggestion
This package provides example server and client implementations of the [Foxglove WebSocket protocol](https://github.com/foxglove/ws-protocol).
```

> For all possible actions, run the following command:

Reviewer: ```suggestion
To list all possible actions, run the following command:
```

```
$ npx @foxglove/ws-protocol-examples --help
```

## Example servers

Run the `sysmon` example server, which uses JSON-encoded data to transmit system monitoring information like memory and CPU usage:

```
$ npx @foxglove/ws-protocol-examples sysmon
```

Run the `image-server` example server, which uses JSON-encoded data to transmit images:

```
$ npx @foxglove/ws-protocol-examples image-server
```

> You must close out each server (`control` + `c`) before starting up the other.

Reviewer: ```suggestion
Note: You must exit each server (<kbd>control</kbd> + <kbd>C</kbd>) before starting up another.
```

To see data from either server, open [Foxglove Studio](https://studio.foxglove.dev?ds=foxglove-websocket&ds.url=ws://localhost:8765/) with a Foxglove WebSocket connection to `ws://localhost:8765/`:

> <img width="500" alt="Foxglove Studio displaying memory and CPU usage from the system monitor example" src="https://user-images.githubusercontent.com/14237/145313065-85c05645-6b29-4eb2-a498-849c83f8792d.png">

Reviewer: Hm, just realized I put a screenshot of the image server in the blog post but wonder if we should do it here too... Here's one if you want it: https://user-images.githubusercontent.com/14237/146500927-4a1408c7-0725-49e7-8185-71b0280c0a8b.png
